service/index.js

```diff
@@ -0,0 +1,56 @@
+const { ObjectId } = require("mongoose").Types;
+const Contact = require("./schemas/contacts");
+
+const getAllContacts = () => Contact.find({}).lean();
+
+const getSingleContact = (contactId) => {
+  let objectIdContact;
+  try {
+    objectIdContact = ObjectId(contactId);
+  } catch (err) {
+    return null;
+  }
```

Comment:
mongoose'owe `ObjectId` moze wyrzucic blad? o_o

Reply:
To jest mapowanie na ObjectId, aby w przypadku, gdy będzie podane ID niezgodne z przyjętą przez MongoDB strukturą nie wyrzuciło błędu. 

Przykład:
W przypadku usunięcia tego mapowania jeżeli np. podany do ID ciąg znaków będzie za krótki lub za długi to zapytanie wyrzuci błąd serwera (error 500):

{
    "message": "Cast to ObjectId failed for value \"niezgodnyCiagZnakowDlaId\" (type string) at path \"_id\" for model \"contact\"",
    "data": "Internal Server Error"
}

Po wprowadzeniu mapowania na ObjectId odpowiedź jest poprawna odpowiedź dla błędu (error 404):

{
    "message": "Contact not found"
}

Reply:
Ok. To sie zgadza z tym, co dostaje od innych. Ale czy potrzebujesz tutaj `try .. catch`? Zwlaszcza, ze `catch` zwraca `null`.

Reply:
null jest zwracane ponieważ funkcja ta jest wywoływana w kontrolerze w postaci wyrażenia funkcyjnego.

W związku z tym jeżeli przekazane do funkcji contactId, nie bedzie zgodne z wymaganym przez MongoDB (tj. w postaci string'u 12 bajtów lub string'u 24 znaków szesnastkowych lub jako liczba całkowita) to wtedy wyrażenie funkcyjne zwraca błąd servera o nieprawidłowym formacie.

Natomiast gdy wykonamy try...catch z mapowaniem contactId przekazywanym do funkcji getSingleContact, zmienna contact w przypadku zlego formatu dostaje wartość null (czyli pusto - brak obiektu kontaktu dla tego ID), a jeśli ID jest w porządku to contact jest po prostu obiektem kontaktu.

W związku z powyższym uważam ze słusznie zastosowałem try...catch dla mapowania Id.

controllers/contactsCtrl.js
`const contact = await service.getSingleContact(contactId);`

service/index.js
```
const getSingleContact = (contactId) => {
  let objectIdContact;
  try {
    objectIdContact = ObjectId(contactId);
  } catch (err) {
    return null;
  }

  return Contact.findOne({ _id: objectIdContact }).lean();
};
```